src/rust/engine/Cargo.lock

```diff
@@ -4,9 +4,9 @@ version = 3
 
 [[package]]
 name = "addr2line"
-version = "0.20.0"
+version = "0.21.0"
```

Comment:
It looks like this change has bumped a whole lot of unrelated(?) dependency versions, which seems potentially unhelpful/risky to lump in with this change. (I don't know what Pants' risk appetite is for whole sale changes like this.)

I _think_ the one way to preserve the old deps is to restore this lock file back to `main`, and then run `./cargo update -p temp-env` to only "update" that package. Alternatively, I think changing `Cargo.toml` and then running any command (e.g. `./cargo check`) will do the "minimal" update to the lock file automatically, unlike running the default `./cargo update` which does a full update.